cmd/kubeadm/app/preflight/checks.go

```diff
@@ -986,6 +989,9 @@ func RunJoinNodeChecks(execer utilsexec.Interface, cfg *kubeadmapi.JoinConfigura
 				HTTPProxyCheck{Proto: "https", Host: ipstr},
 			)
 			if ip := netutils.ParseIPSloppy(ipstr); ip != nil {
+				if netutils.IsIPv4(ip) {
+					checks = addIPv4Checks(checks)
+				}
```

Comment:
Emm... How about DualStack? We need to check both IPv4 and IPv6.
We need a way to check if it's SingleStack v6, SingleStack v4 or DualStack.

Reply:
The current logic should take care of it. as the checks get appended and they don't overwrite.
So if you have IPv4 single stack "IsIPv4" is true and "IsIPv6" is false. So you only get the IPv4 checks.
If you have IPv6 single stack "IsIPv4" is false and "IsIPv6" is true. So you only get the IPv6 checks.
If you have dualstack "IsIPv4" is true and the IPv4 checks are added. But also "IsIPv6" is true and the IPv6 checks are added as well.

Reply:
thanks @agowa338 for the comments, I think @SataQiu 's comments is valid when it comes to the case of dual-stack, as it  originally checks the `cfg.LocalAPIEndpoint.AdvertiseAddress` only which is not enough.